### PR TITLE
[RDY] Merge 'vimdiff' with 'nvim -d'

### PR DESCRIPTION
**Pull request by ghost**
As discussed in #1646, this merges `vimdiff` with `nvim -d`. It's worth mentioning that `vimdiff` is the exact same thing as `nvim -d` (the former is just a symlink).

### CODE REVIEW

**Comment by fmoralesc** [comment]
:+1:

**Comment by fwalch** [comment]
I'd vote for keeping the symlink handling (https://github.com/neovim/neovim/issues/1646#issuecomment-66891125), what do you think?

**Comment by Pyrohh** [comment]
I'd vote for removing it, for the reasons stated by @fmoralesc in the last post of that thread. As you said, a tiny shell script could be used instead. If we're going to remove duplicate interfaces from the docs, then we might as well remove their (limited) references in the code as well.

**Comment by Pyrohh** [comment]
Oops, sorry.

File: src/nvim/version.c [hunk 1]
```diff
@@ -57,7 +57,6 @@ static char *(features[]) = {
   "+cursorbind",
   "+cursorshape",
   "+dialog_con",
-  "+diff",
```
Comment (fmoralesc) [inline src/nvim/version.c]: I don't think this should be removed, some plugin code might want to check on `has('diff')`.

Comment (Pyrohh) [inline src/nvim/version.c]: This is solely for `--version` and `:version` output.

`has('feature')` is handled here https://github.com/neovim/neovim/blob/master/src/nvim/eval.c#L9892

Comment (fmoralesc) [inline src/nvim/version.c]: Ah, OK. Please disregard.

Comment (Hinidu) [inline src/nvim/version.c]: Actually there are plenty vim features which are non-optional in nvim, so perhaps we should do something with all such features in that list. I'm not sure that removing just that one is a good idea.

Comment (Pyrohh) [inline src/nvim/version.c]: @Hinidu I just wanted to keep this PR well contained, since I already removed all references in the docs to `+diff`. I do agree with you though, and I already have a feature branch for that here:
https://github.com/Pyrohh/neovim/commits/cleanup-version-output

The output might need some work though, as it looks a bit odd/incomplete:
```
NVIM 0.0.0-alpha+201501201544 (compiled Jan 20 2015 15:44:59)
Commit: 144c3fc0587ead7f51b9c02438f2a10e5747776c
Build type: Debug
Compilation: /usr/bin/cc -Wconversion -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -I/home/michael/src/neovim/build/config -I/home/michael/src/neovim/src -I/home/michael/src/neovim/.deps/usr/include -I/home/michael/src/neovim/.deps/usr/include -I/home/michael/src/neovim/.deps/usr/include/luajit-2.0 -I/usr/include -I/home/michael/src/neovim/build/src/nvim/auto -I/home/michael/src/neovim/build/include
Compiled by michael@orchid  Features included (+) or not (-):
+acl      +iconv    +terminfo 
   system vimrc file: "$VIM/nvimrc"
     user vimrc file: "$HOME/.nvimrc"
 2nd user vimrc file: "~/.nvim/nvimrc"
      user exrc file: "$HOME/.exrc"
  fall-back for $VIM: "/usr/local/share/nvim"
```

Comment (Pyrohh) [inline src/nvim/version.c]: Also worth mentioning is that I might have removed too many features from that branch, but my understanding is that platform parity is a big goal (hence the removal of the `+mouse_*` and X11 related items.

Comment (fmoralesc) [inline src/nvim/version.c]: Perhaps `Features included (+) or not (-):` could be changed for `Includes support for:`.

Also, a list of removed features would be good to have, to make clean how nvim differs from vim: `-clipboard`, `-clientserver` `-cryptv`, etc. Just as an high level overview, there is no point in being too detailed.

**Comment by Pyrohh** [comment]
I agree, assuming you're not talking about putting such a list in the `--version` output. I think a good idea would be to point users to something like `runtime/doc/nvim_diff.txt` via README.md and `runtime/doc/nvim_intro.txt`,  so it could be viewed on Github and in nvim. I know the wiki page exists, but it's not very comprehensive.

Then again, we could just update the wiki page :)

File: src/nvim/version.c (hunk 1 above)
Comment (Pyrohh) [inline src/nvim/version.c]: (I accidentally replied in the main thread)

**Comment by Pyrohh** [comment]
Squashed the doc changes, rebased and repushed.

File: runtime/doc/diff.txt [hunk 2]
```diff
@@ -1,39 +1,31 @@
...
-This file describes the |+diff| feature: Showing differences between two,
-three or four versions of the same file.
+							*diff* *diff-mode*
+This file describes |diff-mode|, which shows the differences between two,
+three, or four versions of the same file.
 
 The basics are explained in section |08.7| of the user manual.
 
-1. Starting diff mode		|vimdiff|
+1. Starting diff mode		|starting-diff-mode|
```
Comment (justinmk) [inline runtime/doc/diff.txt]: How about using `|diff-mode|` here, and move the tag moved from line 7

Comment (Pyrohh) [inline runtime/doc/diff.txt]: I didn't do that because I thought it would cause entering `:h diff-mode` to bring you to the top of the index, as opposed to right above it (line 8). What do you think?

Comment (justinmk) [inline runtime/doc/diff.txt]: I didn't realize the old tag pointed right above this TOC entry, that's pretty silly. 

But, currently `|starting-diff-mode|` points to nowhere, it doesn't have a tag (a tag is defined with `*`). So just delete `starting-` and then it's fine.

Comment (Pyrohh) [inline runtime/doc/diff.txt]: > (a tag is defined with *)

Ah, will do.

**Comment by Hinidu** [comment]
> I think a good idea would be to point users to something like runtime/doc/nvim_diff.txt via README.md and runtime/doc/nvim_intro.txt

I also thought that the good option is to remove this information from --version output, but provide it somewhere in the docs.

**Comment by Pyrohh** [comment]
> I also thought that the good option is to remove this information from --version output, but provide it somewhere in the docs.

Indeed. I might as well do that along with the `--version` cleanup, if only a stub.

**Comment by Pyrohh** [comment]
Marking as ready, feel free to remove it if I missed something.

**Comment by Pyrohh** [comment]
Rebased and squashed.

File: runtime/doc/diff.txt [hunk 3]
```diff
@@ -1,39 +1,31 @@
...
-between the arguments. >
-
-	vimdiff file1 file2 [file3 [file4]]
-
-This is equivalent to: >
+To start editing in diff mode, start Nvim with "nvim -d".  This starts Nvim
+and sets up for viewing the differences between the arguments. >
 
-	vim -d file1 file2 [file3 [file4]]
+    $ nvim -d file1 file2 [file3 [file4]]
```
Comment (justinmk) [inline runtime/doc/diff.txt]: `$` is not worth the space, and may confuse Windows users. Plus it will cause the rest of the documentation to be inconsistent.

File: runtime/doc/diff.txt [hunk 4]
```diff
@@ -1,39 +1,31 @@
...
 
-You may also use "gvimdiff" or "vim -d -g".  The GUI is started then.
-You may also use "viewdiff" or "gviewdiff".  Vim starts in readonly mode then.
-"r" may be prepended for restricted mode (see |-Z|).
+In addition to the |-d| argument, |-R| and |-Z| may be used for restricted
+mode and read only mode respectively.
```
Comment (justinmk) [inline runtime/doc/diff.txt]: `-R` and `-Z` do not match the order of "respectively".

File: runtime/doc/usr_28.txt [hunk 5]
```diff
@@ -376,7 +376,7 @@ More about folding by expression in the reference manual: |fold-expr|
 *28.9*	Folding unchanged lines
 
 This is useful when you set the 'diff' option in the same window.  The
-|vimdiff| command does this for you.  Example: >
+|-d| argument does this for you.  Example: >
```
Comment (justinmk) [inline runtime/doc/usr_28.txt]: In most of the documentation (and general convention) it is more precisely call an "option".

File: runtime/doc/diff.txt (hunk 4 above)
Comment (Pyrohh) [inline runtime/doc/diff.txt]: :+1:

File: runtime/doc/diff.txt (hunk 3 above)
Comment (Pyrohh) [inline runtime/doc/diff.txt]: :+1: Didn't think about the windows users.

File: runtime/doc/usr_28.txt (hunk 5 above)
Comment (Pyrohh) [inline runtime/doc/usr_28.txt]: I could say `command line option`, to avoid mixing with vim's notion of internal options (e.g. `option`).

Comment (justinmk) [inline runtime/doc/usr_28.txt]: too wordy. In context, there is little risk of confusion, an example follows immediately.

**Comment by Pyrohh** [comment]
Pushed the suggestions.

**Comment by justinmk** [comment]
merged and made some small changes. Thanks!
